components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -1,30 +1,73 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
+import {UserProfile, UserNotifyProps} from 'mattermost-redux/types/users';
+
+import {updateMe} from 'mattermost-redux/actions/users';
+
 import Constants, {NotificationLevels} from 'utils/constants';
-import * as Utils from 'utils/utils.jsx';
-import SettingItemMax from 'components/setting_item_max.jsx';
+import * as Utils from 'utils/utils';
+import SettingItemMax from 'components/setting_item_max';
 import SettingItemMin from 'components/setting_item_min';
 
-import DesktopNotificationSettings from './desktop_notification_settings.jsx';
+import DesktopNotificationSettings from './desktop_notification_settings';
 import EmailNotificationSetting from './email_notification_setting';
-import ManageAutoResponder from './manage_auto_responder.jsx';
+import ManageAutoResponder from './manage_auto_responder';
+
+export interface Props {
+    user: UserProfile & {
+        notify_props: UserProfile['notify_props'] & {
+            auto_responder_active?: 'true' | 'false';
+            auto_responder_message?: string;
+        };
+    };
+    updateSection: (section: string) => void;
+    activeSection: string;
+    closeModal: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    collapseModal: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+    sendPushNotifications: boolean;
+    enableAutoResponder: boolean;
+    actions: {
+        updateMe: typeof updateMe;
```

Comment:
This type is incorrect since `updateMe` would've been wrapped with `bindActionCreators` by this point. I believe the correct type of this would be `(UserProfile) => Promise<ActionResult>`